### ISSUE TITLE
Importing Over 100MB Evernote ENEX Server Stops Responding and When it Loads its Super Slow

### ISSUE THREAD

**Issue by aashish108** [TriliumNext/Trilium#1366]
Hi there, I try importing said evernote notebook, it seems to work but I get error notifications like:

```
Error when calling POST tree/load: timeout - timeout
```

And eventually the server stops responding. After a while the server responds and Trilium loads but Trilium is now super slow to the point of its not really loading/working! I might have to restart the docker image.

What can I do to improve performance pls?
Edit: Eventually, after several minutes performance returns to normal.

Also, a side note, I see a text called advertisement. Are there ads on Trilium? (My Pihole could be blocking them, not sure).

## Info
Version 0.45.2
[trilium-2020-11-01.log](https://github.com/zadam/trilium/files/5470836/trilium-2020-11-01.log)
Using Trilium on Pi :)

Thanks

**Comment by zDEFz** [comment]
related - importing big archives locally into the trillium client throws an Javascript out of range error.

**Comment by zadam** [comment]
Trilium sort of getting overwhelmed by big imports have been already noticed before, but it's difficult to fix this without having ENEX to reproduce this ... :(

As to your next question, there's no advertisement in Trilium.

**Comment by jkaplon** [comment]
Importing my ~150MB ENEX was definitely slow.
My web-server install of Trilium was never able to complete the import, the logs were swamped w/`SLOW QUERY` messages and some image-shrink errors.
I had better luck using the desktop/electron build on a spare PC, but it still took 10-15min to finalize.

This was a few months ago, but I think I remember the process using lots of disk I/O.
If your raspberry-pi is as slow as mine w/disk operations, I'm sure it's not doing you any favors.
My advice is to run the import on a PC w/more resources, then use Trilium's sync feature to get everything over onto the Pi.

**Comment by aashish108** [comment]
Thanks @jkaplon, I'll try that.

@zadam I can donate a 457MB ENEX file if you want? No personal info in it, just research on games lol. Let me know if you want to.

**Comment by zadam** [comment]
That would be highly appreciated. If you can host it somewhere and send me a link to zadam.apps@gmail.com.

Thanks!

**Comment by aashish108** [comment]
I shared the file to you last night @zadam, just fyi here . I hope it was ok?

Thanks.

**Comment by zadam** [comment]
Yes, I received the file and will use it to test the import. Thanks.

**Comment by meichthys** [comment]
[Trilium has entered maintenance mode](https://github.com/zadam/trilium/issues/4620#issuecomment-2094495698). Future enhancements will be addressed in TrilumNext: https://github.com/TriliumNext/trilium/issues/4990

**Comment by meichthys** [comment]
Related to:
https://github.com/TriliumNext/Trilium/issues/776
https://github.com/TriliumNext/Trilium/issues/1133
https://github.com/TriliumNext/Trilium/issues/1423
https://github.com/TriliumNext/Trilium/issues/3616

**Comment by meichthys** [comment]
Closing in favor of #1423
